services/twilioService.js

```diff
@@ -0,0 +1,225 @@
+      return clientInit;
+    }
+
+    return false;
+  }
+
+  async getTwilioToken() {
+    const localToken = JSON.parse(this.hmToken);
+    const requestUri = `${API_URL}/chat/token/${localToken._id}?token=${localToken.api_key}`;
+    // TODO: add device ID
+    const bodyData = { device: 'test' };
+    const response = await fetch(requestUri, { method: 'post', body: JSON.stringify(bodyData), headers: { 'Content-Type': 'application/json' } });
+    const responseJson = await response.json();
+    const twilioToken = responseJson.chat_token;
+    return twilioToken;
+  }
+
+  getChannelName(contact) {
+    const localToken = JSON.parse(this.hmToken);
+    // const contact = this.props.navigation.state.params.mentee._id;
+    return localToken._id > contact ? `${localToken._id}.${contact}` : `${contact}.${localToken._id}`;
+  }
+
+  async updateMessages(contact) {
+    const messages = await this.getRawMessages(contact);
+    await this.messageService.updateLocalMessageState(contact, messages);
+  }
+
+  async getRawMessages(contact) {
+    try {
+      const c = await this.channels[contact];
+      return c.getMessages();
+    } catch (e) {
+      // TODO: add sentry logging
+    }
+    return null;
+  }
+
+  async tryInviteContact(contact) {
+    try {
+      // TODO: if the user isn't found, or if the user trying to invite doesn't have permission, we still
+      //  end up showing the error screen, and not catching the failure...
+      const c = await this.channels[contact];
+      c.invite(contact);
+    } catch (e) {
+      // TODO: add sentry logging
+    }
+  }
+
+  async createChannelWithUser(contact) {
+    const channelName = this.getChannelName(contact);
+    return this.chatClient.createChannel({
+      uniqueName: channelName,
+      friendlyName: channelName,
+    });
+  }
+
+  async initChatClient(token) {
+    try {
+      const chatClient = await TwilioChatClient.create(token, { logLevel: 'info' });
+      this.chatClient = chatClient;
+
+      this.chatClient.on('tokenAboutToExpire', () => {
+        this.getTwilioToken()
+          .then(newData => this.chatClient.updateToken(newData))
+          .catch(
+            // TODO: add sentry logging
+          );
+      });
+
+      // Listen for new invitations to your Client
+      this.chatClient.on('channelInvited', (channel) => {
+        // Join the channel that you were invited to
+        // TODO: add a call to the backend to check if we should be joining this channel
+        channel.join();
+      });
+
+      // this.channel = await this.getChannelForChat();
+      // await this.getMessage();
+      // TODO: Decide which of these callbacks we need
+      // this.subscribeToAllChatClientEvents();
+      return true;
+    } catch (e) {
+      // TODO: add sentry logging
+      return false;
+    }
+  }
+
+  /*
+  async subscribeToAllChatClientEvents() {
+    // This function is not currently called. We should decide which callbacks we want to handle, and how
```

Comment:
Should we provide a link to the docs that list these listening events, instead of providing this big block comment? What if more events get added?

Reply:
Definitely, good call. Added in next push